karapace/utils.py

```diff
@@ -48,3 +50,98 @@ def json_encode(obj, *, compact=True, sort_keys=True, binary=False):
         default=default_json_serialization
     )
     return res.encode("utf-8") if binary else res
+
+
+class Result:
+    def __init__(self, status, json_result, headers=None):
+        # We create both status and status_code so people can be agnostic on whichever to use
+        self.status_code = status
+        self.status = status
+        self.json_result = json_result
+        self.headers = headers if headers else {}
+
+    def json(self):
+        return self.json_result
+
+    def __repr__(self):
+        return "Result(status=%d, json_result=%r)" % (self.status_code, self.json_result)
+
+    def ok(self):
+        return self.status_code == 200
+
+
+class Client:
+    def __init__(self, server_uri=None, client=None):
+        self.server_uri = server_uri
+        self.session = requests.Session()
```

Comment:
thinking this was here so we can weight async vs sync implementation for all clients and components that depend on them. It's there partly there because i think the jury is still out on that one and because i did not really check weather removing it will actually break existing tests / functionality. On a more abstract note i'm pretty much trying to touch as few pieces as possible since fixing components that are not broken in a PR that addresses other issue is an easy way to get sidetracked

Reply:
Yes indeed removing requests here would break our remote target tests. I would also keep this file unchanged and just moved to a new place for now as it's not new code for this PR. We can come back to improving the client later.